apps/docs/src/content/reference/extras/instanced-sprite.mdx

```diff
@@ -0,0 +1,454 @@
+	marking the start and end frames of the animation.
+
+#### Typesafety
+
+For improved developer experience when working with TypeScript, it is strongly recommended to use `as const` assertion in
+combination with `satisfies SpritesheetMetadata`. This approach not only ensures compliance with the `SpritesheetMetadata`
+type but also enables autocompletion for animation names within utility functions, which is highly recommended:
+
+
+### `buildSpritesheet()`
+
+`buildSpritesheet()` is a utility function for building a final texture and spritesheet object from a provided `SpritesheetMetadata` object or external source.
+Each `buildSpritesheet` method return an Promise that and has to be awaited. Promise returned by each method contains an object with a `spritesheet`
+ready for use in `<InstancedSprite>`.
+
+#### buildSpritesheet().from(meta: SpritesheetMetadata)
+Other than `spritesheet` promise, it also returns a `useInstancedSprite` hook. This hook can be enhanced with
+extra typescript support for autocompletion of animation names as such:
+
+
+```ts
+ const meta = [
+    {
+      url: '/textures/sprites/cacodaemon.png',
+      type: 'rowColumn',
+      width: 8,
+      height: 4,
+      animations: [
+        { name: 'fly', frameRange: [0, 5] },
+        { name: 'attack', frameRange: [8, 13] },
+        { name: 'idle', frameRange: [16, 19] },
+        { name: 'death', frameRange: [24, 31] }
+      ]
+    }
+  ] as const satisfies SpritesheetMetadata
+
+  const result = buildSpritesheet.from<typeof meta>(meta)
+```
+
+![Tree sprite atlas](/images/docs/extras/instanced-sprite/useInstancedSpriteAutocomplete.png)
+
+#### buildSpritesheet().fromAseprite(asepriteDataUrl: string, spriteImageUrl: string)
+
+Similar to above, but it parses the Aseprite metadata json into the correct format.
+Does not provide any additional utilities.
+
+### Examples
+
+#### Multiple animations
+
+```ts
+const demonSpriteMeta = [
+    {
+      url: '/textures/sprites/cacodaemon.png',
+      type: 'rowColumn',
+      width: 8,
+      height: 4,
+      animations: [
+        { name: 'fly', frameRange: [0, 5] },
+        { name: 'attack', frameRange: [8, 13] },
+        { name: 'idle', frameRange: [16, 19] },
+        { name: 'death', frameRange: [24, 31] }
+      ]
+    }
+  ] as const satisfies SpritesheetMetadata
+```
+
+#### Multiple files
+
+```ts
+const goblinSpriteMeta = [
+    {
+      url: '/textures/sprites/goblin/Attack.png',
+      type: 'rowColumn',
+      width: 8,
+      height: 1,
+      animations: [{ name: 'attack', frameRange: [0, 7] }]
+    },
+    {
+      url: '/textures/sprites/goblin/Death.png',
+      type: 'rowColumn',
+      width: 4,
+      height: 1,
+      animations: [{ name: 'death', frameRange: [0, 3] }]
+    },
+    {
+      url: '/textures/sprites/goblin/Idle.png',
+      type: 'rowColumn',
+      width: 4,
+      height: 1,
+      animations: [{ name: 'idle', frameRange: [0, 3] }]
+    }
+  ] as const satisfies SpritesheetMetadata
```

Comment:
Why does it need to be `as const satisfies SpritesheetMetadata`? Why not `const goblinSpriteMeta: SpritesheetMetadata = [...]`?